### ISSUE TITLE
[MRP][11.0] tracked materials Quantity Done can't be edited on MO

### ISSUE THREAD

**Issue by oerp-odoo** [odoo/odoo#26510]
**Impacted versions:**
11.0 (might be others too)
**Steps to reproduce:**
Create manufacturing order with raw material that has tracking enabled (lot or serial, doesn't matter which). After creating it, press button 'Unlock'. Press Edit button to edit MO. Open raw move line with that material. 
**Current behavior:**
Quantity Done will be in readonly mode.

**Expected behavior:**
As long as tracked material does not have `finished_lots_exist` as `True`, it should allow editing it.
To me it looks like domain was formed incorrectly, so now it is always in readonly mode. Or is it really intended to have `OR` between all three checks?

Correct domain should be either:
`attrs="{'readonly': ['|', ('is_locked', '=', True), ('finished_lots_exist', '=', True), ('has_tracking', '!=', 'none')]}"`
or even simpler:
`attrs="{'readonly': ['|', ('is_locked', '=', True), ('finished_lots_exist', '=', True)]}"`

Because `finished_lots_exist` should could be true for tracked products anyway, right?

P.S. if such domain is used elsewhere, then it probably needs to be adjusted too.

Video/Screenshot link (optional):
![screenshot from 2018-08-22 10-55-55](https://user-images.githubusercontent.com/7812986/44450588-06d79980-a5fa-11e8-893e-526ae837a9aa.png)

**Comment by oerp-odoo** [comment]
@pedrobaeza oh you are right, forgot all about those lines. Though why it is needed to check if there are finished lots? If tracked product quantity is always readonly anyway? That check will do nothing (I mean `('finished_lots_exist', '=', True)`)

**Comment by pedrobaeza** [comment]
Well, I think the intention of that part if for avoiding to waste traceability from raw lot to finished lot changing the quantities.

**Comment by oerp-odoo** [comment]
As I understand if `has_tracking != 'none'`, then that part (where finshed lots is checked) is useless, because that field will always be readonly anyway. Unless I don't see something here?

I mean you won't be able to change quantities if it is raw lot or finished lot.

**Comment by pedrobaeza** [comment]
But the 3 values should be there for this being editable. I don't understand where do you want to get.

**Comment by oerp-odoo** [comment]
What I meant, domain could be simplified as this `['|', ('is_locked', '=', True), ('has_tracking', '!=', 'none')]`. It then should be editable same way as it is now, but without that extra check. Unless lots are somehow used for products that are not tracked?

**Comment by pedrobaeza** [comment]
No, as said, you shouldn't change the quantity if there are finished lots.

**Comment by oerp-odoo** [comment]
I understand that:). Thats not the point here now (as I said I forgot about those stock move lines). What I mean, if product is tracked, you can't change quantity anyway. So if there are finished lots or not, it does not matter.

Its like to say we only: 
```python
'buy cars' or 'buy cars with four wheels' 
```
This `four wheels` condition is sub condition of `cars`, so it is redundant. I mean if we buy cars, then we also buy cars that have four wheels too. Don't know if I made my point about that domain understandable..:)

**Comment by pedrobaeza** [comment]
OK, got it. Yes, it's redundant, but it doesn't hurt either. You can propose its removal on master branch through a PR (this kind of change is not allowed in stable branches, as it doesn't fix a bug, and the ratio risk/benefit is low).

**Comment by oerp-odoo** [comment]
Yes it could be useful only for refactoring purposes and to make it more clear about that domain intentions
